.github/workflows/fast_testing.yaml

```diff
@@ -39,10 +40,49 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Install tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool != 'debug'
         uses: tarantool/setup-tarantool@v1
         with:
           tarantool-version: ${{ matrix.tarantool }}
 
+      - name: Get last commit from Tarantool
+        if: matrix.tarantool == 'debug'
+        id: get-tnt-commit
+        run: |
+          commit_hash=$(git ls-remote https://github.com/tarantool/tarantool.git HEAD | head -c 8)
+          echo "::set-output name=hash::${commit_hash}"
+        shell: bash
+
+      - name: Cache tarantool build
+        if: matrix.tarantool == 'debug'
+        id: cache-tnt-debug
+        uses: actions/cache@v3
+        with:
+          path: ~/tnt-build/
+          key: cache-tnt-${{ matrix.tarantool }}-${{ steps.get-tnt-commit.outputs.hash }}
+
+      - name: Build tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool == 'debug' && steps.cache-tnt-debug.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get -y install git build-essential cmake make zlib1g-dev \
+            libreadline-dev libncurses5-dev libssl-dev \
+            libunwind-dev libicu-dev python3 python3-yaml \
+            python3-six python3-gevent
+          cd ~/
+          git clone https://github.com/tarantool/tarantool.git --recursive
+          cd tarantool
+          git submodule update --init --recursive
+          mkdir build && cd build
+          cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=~/tnt-build/
+          make
+          sudo make install
+          sudo cp extra/dist/tarantoolctl ~/tnt-build/bin/
+
+      - name: Install tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool == 'debug'
+        run: |
+          sudo cp -rvaP ~/tnt-build/* /usr/local/
+
       - name: Clone the module
         uses: actions/checkout@v2
```

Comment:
Maybe we can also bump actions versions to currently relevant versions while we are here. AFAIK it's v3 for now. Of course, in separate commit.

Reply:
A good idea. Please, see the first commit.